guardian/editions
# Make api url asynchronous when fetching editions list

## Summary
When testing the special Earth Edition on CODE proof we noticed that occasionally the special edition would be missing from the editions menu and the app would not reopen on the Earth Edition after the app was killed. 

When investigating the issue I found it was not that we didn't cache the special editions but rather there was sometimes a race condition on the `apiUrl` which meant that it could be replaced by the prod API endpoint even when the app was pointing at an alternative endpoint in settings. This meant that the fetched and cached list of editions was being replaced with the prod editions list which did not yet contain the special edition. 

This PR makes the fetching of the `apiUrl` blocking by accessing it through async storage. I found this mechanism was already being used in `fetchIssueWithFrontsFromAPI` for fetching issues and so didn't need to add any other complexity. 



[**Trello Card ->**](https://trello.com/c/WC3LS9FS/1596-cache-special-editions-for-edition-switcher)
[**Trello Card ->**](https://trello.com/c/k7tA3Tr9/1598-special-editions-edition-logic-in-case-the-app-is-killed)

## Test Plan
1. Point to a different endpoint - eg. CODE Proof 
2. Open the Earth special edition 
3. Kill and restart the app multiple times 
4. Observe Earth special edition remains open and in editions menu

Also worth testing that everything works ok when pointing to production.
